app/models/open_studios_event.rb

```diff
@@ -44,6 +48,17 @@ def self.current
 
   private
 
+  SPECIAL_EVENT_FIELDS = %w[special_event_start_time special_event_end_time special_event_start_date special_event_end_date].freeze
+  def generate_special_event_time_slots
+    return if (changed & SPECIAL_EVENT_FIELDS).blank?
+
+    time_slots = (special_event_start_date.day..special_event_end_date.day).map.each_with_index do |_day, idx|
```

Comment:
This is interesting. If I'm reading this correctly, you don't care what the `_day` value is, you just want to map over the number of days. I'm curious if this would be more readable.
```ruby
number_of_event_days = (special_event_end_date - special_event_start_date).to_i
time_slots = (0..number_of_event_days).flat_map do |idx| # which can now be renamed to day_number or something
  ........
end
```
It's essentially the same thing, but it takes out the `each_with_index`. This also will fix any issues that will occur when an event starts on the 28th of one month and ends on the 3rd of the next month. I know that's not relevant for this first virtual open studios, but it was just something I was thinking about.

Reply:
your code is better but still i think problematic.  number_event_days is probably now the number of seconds between the events.  But the idea is totally the right direction.

I got lucky.  I should have written harder tests.

I was rushing a bit to get this right because from the db, i'd like to not have to rebuild these fields later and i want to push stuff to prod... a little cart before the horse.

Reply:
Totally my bad- :TIL:  `(time - time).to_i` gives you days :D